lib/watir/attribute_helper.rb

```diff
@@ -38,6 +38,10 @@ def attribute_list
     #  @return [$1] value of $3 property
     #
     def attribute(type, method, attr)
+      if %i(thead tfoot).include? method
```

Comment:
Then those attribute methods should be be generated in the first place. Do something like this to achieve that:

```diff
diff --git a/lib/watir/generator/html/generator.rb b/lib/watir/generator/html/generator.rb
index 8a7ed73..8f07633 100644
--- a/lib/watir/generator/html/generator.rb
+++ b/lib/watir/generator/html/generator.rb
@@ -16,7 +16,9 @@ module Watir
       end

       def ignored_attributes
-        %w(cells elements hash rows span text)
+        attributes = %w(elements hash span text)
+        # table elements should not be accessible as attributes
+        attributes += %w(cells tHead tFoot rows)
       end

       def generator_implementation
```